packages/typography/src/Typography.tsx

```diff
@@ -117,33 +100,60 @@ Subtitle.displayName = 'Subtitle';
 /**
  * Body
  */
-type BodyProps = HTMLElementProps<'div'> & {
+
+type BodyFontWeight = 'regular' | 'medium';
+type BodyProps = HTMLElementProps<'p'> & {
   /**
    * font-weight applied to typography element
    * default: `regular`
    */
-  weight?: 'regular' | 'medium';
+  weight?: BodyFontWeight;
+  as?: keyof JSX.IntrinsicElements;
 };
 
-function Body({ children, className, weight = 'regular', ...rest }: BodyProps) {
+const Body: ExtendableBox<BodyProps, 'p'> = ({
+  className,
+  weight = 'regular',
+  as = 'p',
+  ...rest
+}: BodyProps) => {
   const size = useBaseFontSize();
   const body = size === 16 ? typeScale2 : typeScale1;
+  const fontWeights: {
+    [key: string]: {
+      [key in BodyFontWeight]: number;
+    };
+  } = {
+    default: {
+      regular: 400,
+      medium: 500,
+    },
+    strong: {
+      regular: 700,
+      medium: 800,
+    },
+    b: {
+      regular: 700,
+      medium: 800,
+    },
```

Comment:
👍  Nice. Might be cleaner to just use an `if` (e.g. if we want to do the same for h# tags), but not critical